dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt

```diff
@@ -0,0 +1,111 @@
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import kotlin.math.pow
+import kotlin.math.round
+import kotlin.math.sqrt
+import kotlin.system.measureNanoTime
+
+private const val ITEMS_TO_JOINT = 10_000
+private const val REPETITION_COUNT = 10_000
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+@ForgeConfiguration(value = Configurator::class, seed = 0x50c72968d123L)
+internal class JointToStringVsStringBuilderPerformanceTest {
+
+    @Test
+    fun `M be faster than joinToString W buildString`(forge: Forge) {
+        val itemsForJoin = forge.aList(ITEMS_TO_JOINT) { forge.aString() }
+        val jointToStringExecutionTime = mutableListOf<Long>()
+        val buildStringExecutionTime = mutableListOf<Long>()
+
+        var jointToStringResult: String
+        var builderResult: String
+
+        repeat(REPETITION_COUNT) {
+            jointToStringExecutionTime.add(
+                measureNanoTime {
+                    val jointToStringContainer = mutableListOf<String>()
+                    for (item in itemsForJoin) {
+                        jointToStringContainer.add(item)
+                    }
+                    jointToStringResult = jointToStringContainer.joinToString(separator = " ") { it }
+                }
+            )
+
+            buildStringExecutionTime.add(
+                measureNanoTime {
+                    builderResult = buildString {
+                        itemsForJoin.forEach { item -> appendIfNotEmpty(' ').append(item) }
+                    }
+                }
+            )
+
+            assertThat(builderResult).isEqualTo(jointToStringResult) // same result
+        }
+
+        val statisticsReport = (
+            "buildString:\n" +
+                " mean = ${buildStringExecutionTime.mean}\n" +
+                " std = ${buildStringExecutionTime.std}\n" +
+                " cv = ${"%.2f".format(buildStringExecutionTime.cv)}%\n" +
+                " p50 = ${buildStringExecutionTime.percentile(50)}\n" +
+                " p90 = ${buildStringExecutionTime.percentile(90)}\n" +
+                " p95 = ${buildStringExecutionTime.percentile(95)}\n" +
+                " p99 = ${buildStringExecutionTime.percentile(99)}\n" +
+                "\n" +
+                "joinToString:\n" +
+                " mean = ${jointToStringExecutionTime.mean}\n" +
+                " std = ${jointToStringExecutionTime.std}\n" +
+                " cv = ${"%.2f".format(jointToStringExecutionTime.cv)}%\n" +
+                " p50 = ${jointToStringExecutionTime.percentile(50)},\n" +
+                " p90 = ${jointToStringExecutionTime.percentile(90)},\n" +
+                " p95 = ${jointToStringExecutionTime.percentile(95)},\n" +
+                " p99 = ${jointToStringExecutionTime.percentile(99)}\n"
+            )
+
+        println(statisticsReport)
```

Comment:
debugging leftover?

Reply:
The idea was to print this information, but without failed test nobody will read it. Changed to `withFailMessage` instruction